protos/flyteidl/admin/common.proto

```diff
@@ -245,3 +245,14 @@ message AuthRole {
   }
 }
 
+// Specifies how to populate Get{Workflow,Node,Task}ExecutionData responses.
+message GetExecutionData {
```

Comment:
so the reason why I suggested we make this change is because - 

1. Metadata is entirely owned by Flyte. Returning a signed URL is actually a leaky abstraction and forces us to leak the detail that the metadata is stored in S3. A future world, we may store the metadata in DynamoDB for example.
2. One problem with always returning the data is that, sometimes the data is very large and that has problems in FlyteAdmin.

WDYT? if we always return the entire metadata (as literal map) and also the signed url (for backwards compatibility). If the data is larger than some threshold (can be verified using HEAD request), then we only return the signed URL.
Then we start tracking how many times this happens and start figuring out a solution to eliminate that case

Reply:
sure, we're already issuing the HEAD request to capture the data size for the URLBlob portion of the response anyways. I can get rid of the mode and return the data always when it's under a configured limit.

Reply:
I would love if we could get to a solution where we aren't signing URLs if the user doesn't absolutely need to use them. I think from Console's perspective, it would be preferable to see either the data or the URLs. And I would be okay if that's something I have to opt in to with a flag of some sort.
But if that feels like it's over-complicated, or there's no real concern about issuing a signed URL every time we request execution data, please disregard!

Reply:
I don't think including a signed URL adds too much overhead (we have to do the HEAD request either way) so I'm okay returning it by default but if we want to phase it out, happy to make this configurable by a flag

@kumare3 thoughts?

Reply:
so @schottra and @katrogan I think eventually all data that we return should fit within the limits. As we should start offloading lists and long strings should just be outright rejected. So in this scenario, it would be that we always return data inline. SO I would love to move in that direction, this makes it work perfectly in all conditions and hides the complexity in Admin

Reply:
cool, so for the sake of backwards compatibility while we migrate to that model, what if we just always return the signed URL and the full data?